### DISCUSSION TITLE
Tell me what's wrong?

### DISCUSSION THREAD

**Discussion by seoeaa** [safevideo/autollm#210] [Q&A] [answered]
`import os
import gradio as gr
from autollm import AutoQueryEngine, read_files_as_documents

os.environ["OPENROUTER_API_KEY"] = ""

relative_folder_path = "examples/data"
documents1 = read_files_as_documents(input_dir=relative_folder_path , recursive=True)

os.environ["OR_SITE_URL"] = "" # optional
os.environ["OR_APP_NAME"] = "" # optional
os.environ['HUGGINGFACE_API_KEY'] = ""

query_engine = AutoQueryEngine.from_defaults(
    documents=documents1,
    llm_model="openrouter/google/gemini-pro",
    llm_max_tokens=1000,
    llm_temperature=0.1,
    system_prompt="................",
    query_wrapper_prompt="......................",
    enable_cost_calculator=False,
    embed_model='local:intfloat/multilingual-e5-small', #huggingface-cli download multilingual-e5-small
    chunk_size=512,
    chunk_overlap=64,
    context_window=4096,  # Установите это значение, если вам нужно
    similarity_top_k=6,
    response_mode="compact",  # Выберите нужный режим ответа
    refine_prompt=None,  # Установите это значение, если вам нужно
    structured_answer_filtering=False,  # Установите это значение, если вам нужно
    vector_store_type="LanceDBVectorStore",
    lancedb_uri="./.lancedb",
    lancedb_table_name="vectors",
    enable_metadata_extraction=True  
)

query = ".................."
unicode_escape_text = query.encode('unicode_escape').decode()
response = query_engine.query(query)
print(response.response)
def greet(query):
    return query_engine.query(query).response
demo = gr.Interface(fn=greet, inputs="text", outputs="text", title="BOT", description="DES", )
demo.launch(share=True)`

`PS D:\Programm\AI\autollm> & d:/Programm/AI/autollm/myenv/Scripts/python.exe d:/Programm/AI/autollm/app.py
Traceback (most recent call last):
  File "d:\Programm\AI\autollm\app.py", line 3, in <module>
    from autollm import AutoQueryEngine, read_files_as_documents
  File "d:\Programm\AI\autollm\autollm\__init__.py", line 12, in <module>
    from autollm.auto.fastapi_app import AutoFastAPI
  File "d:\Programm\AI\autollm\autollm\auto\fastapi_app.py", line 10, in <module>
    from autollm.serve.utils import load_config_and_initialize_engines, stream_text_data
  File "d:\Programm\AI\autollm\autollm\serve\utils.py", line 7, in <module>
    from autollm.auto.query_engine import AutoQueryEngine
  File "d:\Programm\AI\autollm\autollm\auto\query_engine.py", line 12, in <module>
    from autollm.auto.llm import AutoLiteLLM
  File "d:\Programm\AI\autollm\autollm\auto\llm.py", line 4, in <module>
    from llama_index.llms.base import BaseLLM
ImportError: cannot import name 'BaseLLM' from 'llama_index.llms.base' (D:\Programm\AI\autollm\myenv\lib\site-packages\llama_index\llms\base.py)`

**Comment by fcakyon** [comment]
What are your autollm and llama-index versions @seoeaa ?

**Comment by talhasarit** [comment]
can you please update the `autollm` to latest and try again?

**Comment by talhasarit** [comment, marked as answer]
It should be resolved with 'pip install autollm -U' in latest version, which is 0.1.9
